### PR TITLE
Add go/v3-alpha 

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description of the change:**
- Add go/v3-alpha 
- Add the same sample generated with go/v3-alpha

**Motivation for the change:**
- Support k8s 1.20
- Allows test the projects built with go/v3-alpha and check the diff between v2 X v3-alpha 

**Follow-ups**
- Use samples on CI #4056
- Then, after the above, we can add e2e tests for the v3-alpha sample as well as suggested by @estroz 
- And then, perform the change is the mock as suggested https://github.com/operator-framework/operator-sdk/issues/4234

### CODE REVIEW

File: .gitignore
Comment (estroz) [inline .gitignore:18]: We should combine all `bin/` ignores
```suggestion
*/bin/*
```

File: Makefile
Comment (estroz) [inline Makefile:12]: Since go/v3-alpha isn't the default yet, this should stay the same

File: hack/generate/samples/internal/go/generate_v2.go
Comment (estroz) [inline hack/generate/samples/internal/go/generate_v2.go:31]:
```suggestion
// RunV2 the steps to create the Memcached with Webhooks Go Sample
func (mh *MemcachedGoWithWebhooks) RunV2() {
```

File: hack/generate/samples/internal/go/generate_v3.go
Comment (estroz) [inline hack/generate/samples/internal/go/generate_v3.go:60]:
```suggestion
		"--programmatic-validation")
```

**Review by estroz — CHANGES_REQUESTED** [review]
I'm still hesitant to add v3-alpha samples because it is in alpha, but I guess we do need to test our integrations. We also need e2e tests for v3-alpha.

**Comment by camilamacedo86** [comment]
HI @estroz,

> I'm still hesitant to add v3-alpha samples because it is in alpha, but I guess we do need to test our integrations. We also need e2e tests for v3-alpha.

IMO: We are providing a new option for the users then, we also need to have an example over that. Otherwise, it makes it very harder and increases our effort either to contribute with the further changes and solve the issues raised in the repo. Note as well that: 

- By running the samples we are ensuring at least the mainly commands/scaffolds provided by the tool with the new plugin
- The testdata are important and relevant not only to users. We need the samples to be able to compare v2 X v3 and check the diffs, for we are able to very the further changes in the plugins and scaffolds, have a mock sample to test the projects scaffolds with v3-alpha, help use to built the ansible/helm v2 alpha plugins, check/review and validate the next bumps that affect/change ve3-alpha  and etc. 

c/c @jmrodri @joelanford @varshaprasad96

File: hack/generate/samples/internal/go/generate_v3.go
Comment (camilamacedo86) [reply hack/generate/samples/internal/go/generate_v3.go:60]: Currently, the sample works with `defaulting `. If we change it for v3 we need to change it for v2. Otherwise, we are unable to compare both to check what changed besides increasing the changes/mock/test that cannot be re-used both at this moment. But we can do it in a follow up. Track it in:

**Comment by joelanford** [comment]
Total drive-by: Before we merge this, we need to decide what our support statement will be for v3-alpha projects, and if its anything other than "full support like all other plugins AND full backwards-compatibility guarantees", then we need to very clearly document what the support will be and what guarantees and non-guarantees there are for backward and forward compatibility.

/hold

**Comment by camilamacedo86** [comment]
> I'm worried about telling people they need to use v3-alpha to get access to Kubernetes 1.20 features, and then in the same breath telling them that we can't support them because they're using an alpha feature and that we'll also break their project in the future (e.g. if/when the phase 2 plugins work changes the format of the PROJECT file)

IMO we can: 
- Tell that if users would like to use the 1.20 features than can try out the alpha plugin which is available but it is an alpha implementation which means that can sofer change and etc and that it will ONLY full supported when this plugin version be stable and the be scaffolded by default.

File: hack/generate/samples/internal/go/generate_v3.go
Comment (estroz) [reply hack/generate/samples/internal/go/generate_v3.go:60]: You've specified `--defaulting` twice

**Comment by joelanford** [comment]
Based on last week's discussion, I think we all agreed that kubebuilder's project version 3 and its go/v3 plugin need to be finalized before we will accept them and begin using them in the SDK.

These milestones are prerequisites for scaffolding 1.20-compatible projects because the version of controller-runtime that will support 1.20 (likely v0.8.0) will inherit the breaking changes made in controller-runtime v0.7.0 that are incompatible with the currently supported go.kubebuilder.io/v2 plugin.

**Comment by camilamacedo86** [comment]
Hi @joelanford, 

Yes. I will close this one and then, this PR can be used as a helper for when we are able to do it since here we are generating the sample with v3/go plugin.   

Pinging @estroz @jmrodri and @varshaprasad96 just for let they know that it has what is required for we add the support to the new plugin as it is be stabelized.

**Pull request closed without merging**
